torch/_higher_order_ops/cond.py

```diff
@@ -346,18 +383,24 @@ def cond_autograd(pred, true_fn, false_fn, operands):
         with torch._C._AutoDispatchBelowAutograd():
             return cond_op(pred, true_fn, false_fn, operands)
 
-    (
-        fw_true_graph,
-        fw_false_graph,
-        joint_true_graph,
-        joint_false_graph,
-    ) = create_fw_bw_graph_branches(true_fn, false_fn, *operands)
+    true_bw_fn = create_bw_fn(
+        true_fn,
+        len(operands),
+        torch._C._dispatch_tls_local_include_set(),
+        torch._C._dispatch_tls_local_exclude_set(),
```

Comment:
I'm not completely convinced that these are correct in general, like one could enable autocast in the backward pass but not the forward pass. But that seems rare to me.

Reply:
yeah, we need to figure out a way to pass the tls keyset in bacwkard to autograd.function.backward. For now, the contract we have is that any dispatchkey used in forward.  Since we're materializing the graph, it basically means we defer the materialization until backward is called using the tls dispatch state during forward.